libs/core/overview/components/overview-screens/pure-overview/use-redirect-on-active-device-detached.hook.ts

```diff
@@ -7,30 +7,28 @@ import { useHistory } from "react-router-dom"
 import { useDispatch, useSelector } from "react-redux"
 import { useCallback } from "react"
 import { Dispatch } from "Core/__deprecated__/renderer/store"
-import { isActiveDeviceAttachedSelector } from "Core/device-manager/selectors/is-active-device-attached.selector"
 import { getDevicesSelector } from "Core/device-manager/selectors/get-devices.selector"
 import { deactivateDevice } from "Core/device-manager/actions/deactivate-device.action"
 import {
   URL_DISCOVERY_DEVICE,
   URL_MAIN,
 } from "Core/__deprecated__/renderer/constants/urls"
+import { setSelectDeviceDrawerOpen } from "Core/device-select/actions/set-select-device-drawer-open.action"
 
-export const useHandleActiveDeviceDetached = () => {
+export const useRedirectOnActiveDeviceDetached = () => {
   const history = useHistory()
   const dispatch = useDispatch<Dispatch>()
-  const activeDeviceAttached = useSelector(isActiveDeviceAttachedSelector)
   const devices = useSelector(getDevicesSelector)
 
   return useCallback(() => {
-    if (!activeDeviceAttached) {
-      void dispatch(deactivateDevice())
-      if (devices.length > 1) {
-        history.push(URL_DISCOVERY_DEVICE.availableDeviceListModal)
-      } else if (devices.length === 1) {
-        history.push(URL_DISCOVERY_DEVICE.root)
-      } else {
-        history.push(URL_MAIN.news)
-      }
+    void dispatch(deactivateDevice())
+    if (devices.length > 1) {
+      dispatch(setSelectDeviceDrawerOpen(false))
+      history.push(URL_DISCOVERY_DEVICE.availableDeviceListModal)
+    } else if (devices.length === 1) {
+      history.push(URL_DISCOVERY_DEVICE.root)
+    } else {
+      history.push(URL_MAIN.news)
     }
-  }, [devices, history, dispatch, activeDeviceAttached])
+  }, [devices, history, dispatch])
```

Comment:
This hook not only redirects as name says, but also deactivated device. Maybe it would be good idea to rename it to 'useDeactivateDeviceAndRedirect' or similar?

Reply:
Yes, you're correct... I've been thinking about it, and since there are two of us, I'm going to make a change.

Reply:
https://github.com/mudita/mudita-center/pull/1687/commits/d03a67f37a9df873f450a04e4389a6757bc89eaf